### ISSUE TITLE
Volume node not recognized when porting to another project

### ISSUE THREAD

**Issue by procked** [alexirae/unreal-audio-dsp-template-UE5#1]
Hi!

Thank you so much for the template. It's extremely helpful given the limited documentation for custom nodes.

Cloning the project and building from the .sln file works great, I can see the Volume node from the menu when using the metasound graph from within the built project.

I'm currently trying to transfer this to an existing project (my own plugin). I can't seem to get the node to show up in the metasound graph. I copied the `DSPProcessing` and `MetasoundNodes` folders into my own plugin, built successfully (no errors after fixing the `<CLASS>_API` definition inside `Volume.h` to my own plugin class), but the node doesn't show up in the metasound graph editor. My own plugin is recognized in the `Edit -> Plugins` window in the editor.

This is probably more of a UE question than a metasound question, but there is no information I can find anywhere else. Do you know what the issue could be? I'd greatly appreciate your help!

Thank!

**Comment by alexirae** [comment]
Hi @procked!

I'm glad that the template is being helpful!

Regarding the issue, I remember that I had a similar issue when I created the template, I was missing a call to register the node. 

Check in the template project the file **AudioDSPTemplate.cpp**, the **StartupModule** function and make sure that in your plugin you do this exact same call:
`FMetasoundFrontendRegistryContainer::Get()->RegisterPendingNodes();`

Hopefuly this will fix your issue, otherwise we can keep the ball rolling here!

Best,

**Comment by procked** [comment]
Ahhhhh that was the issue. Thank you so much!!! 

Out of curiosity, where did you figure out that you needed to do that? From the UE source code?

Wish the official documentation was a bit better here, guess that's what we get for using beta/experimental stuff :)

**Comment by alexirae** [comment]
No prob!

Indeed when I started to experiment creating a new node in a custom plugin project I had that issue, I tried several things with no success in order to get the plugin to show up in the Metasound editor then I left the project abandoned for a couple of months.

When I got back into it I started to explore deep into UE metasound code and I noticed that they were registering their nodes in:
`UE_5.0\Engine\Plugins\Runtime\Metasound\Source\MetasoundStandardNodes\Private\MetasoundStandardNodesModule.cpp`

I did the same on my plugin project and baaam! that did the magic :)

I'll add in the GitHub Readme this detail of registering the nodes, I think is extremely important and people needs to know it.

Thanks a lot for finding this out!

Enjoy Metasound nodes development!

Best,
